src/industry_gui.cpp

```diff
@@ -1449,11 +1485,28 @@ class IndustryDirectoryWindow : public Window {
 			cargos.emplace_back(i->produced_cargo[j], i->last_month_production[j], cargo_suffix[j].text, ToPercent8(i->last_month_pct_transported[j]));
 		}
 
-		/* Sort by descending production, then descending transported */
-		std::sort(cargos.begin(), cargos.end(), [](const CargoInfo a, const CargoInfo b) {
-			if (std::get<1>(a) != std::get<1>(b)) return std::get<1>(a) > std::get<1>(b);
-			return std::get<3>(a) > std::get<3>(b);
-		});
+		switch (this->industries.SortType()) {
+			case IDW_SORT_BY_NAME:
+			case IDW_SORT_BY_TYPE:
+			case IDW_SORT_BY_PRODUCTION:
+				/* Sort by descending production, then descending transported */
+				std::sort(cargos.begin(), cargos.end(), [](const CargoInfo a, const CargoInfo b) {
+					if (std::get<1>(a) != std::get<1>(b)) return std::get<1>(a) > std::get<1>(b);
+					return std::get<3>(a) > std::get<3>(b);
+				});
+				break;
+
+			case IDW_SORT_BY_TRANSPORTED:
+				/* Sort by descending transported, then descending production */
+				std::sort(cargos.begin(), cargos.end(), [](const CargoInfo a, const CargoInfo b) {
+					if (std::get<3>(a) != std::get<3>(b)) return std::get<3>(a) > std::get<3>(b);
+					return std::get<1>(a) > std::get<1>(b);
+				});
+				break;
+
+			default:
+				NOT_REACHED();
```

Comment:
You're covering every value of the enum already, so no need for a default block. Compiler will warn if (somehow) another value gets added to the enum but this switch gets forgotten. Would get silently missed (until it causes a crash) otherwise

Reply:
I moved default up and removed NOT_REACHED().